code/common/data.fs

```diff
@@ -43,11 +47,20 @@ let readSnippets () =
 
 let mutable snippets, publicSnippets = readSnippets ()
 
-let loadSnippet id =
-  File.ReadAllText(sprintf "%s/../../data/formatted/%d/0" __SOURCE_DIRECTORY__ (demangleId id))
+let rec private loadSnippetInternal basePath id revision =
+  let id' = demangleId id
+  match Seq.tryFind (fun s -> s.ID = id') publicSnippets with
+  | Some snippetInfo ->
+      match revision with
+      | Latest -> loadSnippetInternal basePath id (Revision (snippetInfo.Versions - 1))
```

Comment:
I see how this works, but I think it would be simpler to do this without recursion. You can probably do just `let r = match revision with Latest -> snippetInfo.Versions - 1 | Revision r -> r` to get the number and then read the file as on the line below.